src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,215 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+    private int halfOfSize = size / 2;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node currentNode = new Node<>(null, value, null);
+            tail = currentNode;
+            head = currentNode;
+            size++;
+        } else if (size > 0) {
+            Node previousNode = head;
+            Node currentNode = new Node(previousNode, value, null);
+            previousNode.next = currentNode;
+            head = currentNode;
+            size++;
```

Comment:
1. This code could be extracted to private method and be used for other cases, like inserting on index. How about `addLast()`?
2. Move common code outside `if/else` block